### DISCUSSION TITLE
Debounce for Async data fetching

### DISCUSSION THREAD

**Discussion by angularui06** [afconsult/quill-mention#300] [Q&A] [answered]
### Requirement: I need to show the suggestions of the user on typing some value after @ using debounce.

`const quill = new Quill("#editor", {
  modules: {
    mention: {
      allowedChars: /^[A-Za-z\sÅÄÖåäö]*$/,
      mentionDenotationChars: ["@", "#"],
      source: async function(searchTerm, renderList) {
        const matchedPeople = await suggestPeople(searchTerm); // **How to use Debounce here?**
        renderList(matchedPeople);
      }
    }
  }
})`

### Expected Solution: 

**_Adding one property in the seetings like **debounceTime: 1000**, will be very helpful._** and  **source** method should 
call after the **debounceTime**

`mention: {
      allowedChars: /^[A-Za-z\sÅÄÖåäö]*$/,
      mentionDenotationChars: ["@", "#"],
      **debounceTime: 1000,**
      source: async function(searchTerm, renderList) {
        const matchedPeople = await suggestPeople(searchTerm); // **How to use Debounce here?**
        renderList(matchedPeople);
      }
    }`

**Comment by aprilmintacpineda** [comment]
Yeah, debounce will be great.

**Comment by Gray-Simon** [comment, marked as answer]
Hi guys,

Here is the solution to the debounce 

 mentionResults$: Observable<any>;
 mentionSubject = new Subject<string>();
 mentionSubscription;

**put this in your constructor:**
 this.mentionResults$ = this.mentionSubject.pipe(
      debounceTime(1000),
      distinctUntilChanged(),
      mergeMap(searchText => this.http.get(api/searchtext) // make your api call)
    );

**then in your source function**
` source: async (searchTerm: string, renderList: Function, trigger: string)`

if (!this.mentionSubscription) {
              this.mentionSubscription = this.mentionResults$.subscribe(res => {
                matches = res;
                renderList(res);
              });
            }
     
 this.mentionSubject.next(searchTerm);
  renderList(matches);
